packages/react-instantsearch-hooks/src/InstantSearchSSRProvider.tsx

```diff
@@ -22,6 +22,12 @@ export function InstantSearchSSRProvider({
   children,
   ...props
 }: InstantSearchSSRProviderProps) {
+  // If there's no server state passed, we don't want to override something
+  // passed from higher in the tree
+  if (Object.keys(props).length === 0) {
+    return <>{children}</>;
```

Comment:
I'm wondering about the implications of having `<InstantSearchSSRContext.Provider>` inconsistently in the React tree. What is the difference with rendering `<InstantSearchSSRContext.Provider value={null}>`?